### ISSUE TITLE
Form with type List and Enum doesn't recognize comma-separated values

### ISSUE THREAD

**Issue by aliev** [tiangolo/fastapi#2364]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [ ] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

I'm trying to create the form with multiple-choice in post request like so:

```Python
class GranTypes(Enum):
    TYPE_AUTHORIZATION_CODE = "authorization_code"
    TYPE_PASSWORD = "password"
    TYPE_CLIENT_CREDENTIALS = "client_credentials"
    TYPE_REFRESH_TOKEN = "refresh_token"


@router.post("/client/create")
async def create_client(grant_types: List[GranTypes] = Form(...)):
    return {'status': 'ok'}
```

### Description

Swagger generated the UI with the ability to multiple-choice of the values:

<img width="1426" alt="Screen Shot 2020-11-17 at 18 56 29" src="https://user-images.githubusercontent.com/407134/99405818-9c138680-2906-11eb-9a34-3ff6f5ee1319.png">

It works when I selecting only one option, but when I selecting many of them, the options are comma-separated for some reason.

Example with one option:

```
curl -X POST "http://127.0.0.1:8000/user/client/create" -H  "accept: application/json" -H  "Content-Type: application/x-www-form-urlencoded" -d "grant_types=password"
```

Response:

```
{
  "status": "ok"
}
```

When multiple options are selected:

```
curl -X POST "http://127.0.0.1:8000/user/client/create" -H  "accept: application/json" -H  "Content-Type: application/x-www-form-urlencoded" -d "grant_types=password,client_credentials"
```

Response:

```
{
  "detail": [
    {
      "loc": [
        "body",
        "grant_types",
        0
      ],
      "msg": "value is not a valid enumeration member; permitted: 'authorization_code', 'password', 'client_credentials', 'refresh_token'",
      "type": "type_error.enum",
      "ctx": {
        "enum_values": [
          "authorization_code",
          "password",
          "client_credentials",
          "refresh_token"
        ]
      }
    }
  ]
}
```

It works as expected with modifying the curl options a bit:

```
curl -X POST "http://127.0.0.1:8000/user/client/create" -H  "accept: application/json" -H  "Content-Type: application/x-www-form-urlencoded" -d "grant_types=password" -d "grant_types=client_credentials"
```

but when I select options via swagger UI for some reason it adds a comma instead of multiple options.

Is it expected behavior? Looks like the bug in Swagger UI

### Environment

* OS: [e.g. Linux / Windows / macOS]: macOS
* FastAPI Version: 0.61.2
* Python Version: Python 3.8.0
* Pydantic version: 1.7.2

**Comment by ycd** [comment]
Hmmm, this is kinda odd, when I try Query it adds them as multiple options,  and generates this URL 

```
/client/create?grant_types=password&grant_types=refresh_token
```


This should be the expected behavior right? I checked the OpenAPI specification and Swagger but couldn't find anything about the string Enums and Form data.

```python
from fastapi import FastAPI, Query

from typing import List  
from enum import Enum 


class GranTypes(str, Enum):
    TYPE_AUTHORIZATION_CODE = "authorization_code"
    TYPE_PASSWORD = "password"
    TYPE_CLIENT_CREDENTIALS = "client_credentials"
    TYPE_REFRESH_TOKEN = "refresh_token"

app = FastAPI()


@app.post("/client/create")
async def create_client(grant_types: List[GranTypes] = Query(...)):
    return {'status': 'ok'}
```

**Comment by kernc** [comment]
Seems more a Pydantic issue, no?
```py
from enum import Enum
from typing import List

from pydantic import BaseModel


class GrantType(str, Enum):
    token = 'token'
    client = 'client'
    ...


class Model(BaseModel):
    grants: List[GrantType]


Model(grants='token,client')  # ValidationError
```

**Comment by kernc** [comment]
Nah, it's Swagger UI bug: https://github.com/swagger-api/swagger-ui/issues/3494. It should be sending multiple separate parts instead of a coalesced one.

**Comment by aliev** [comment]
@kernc thanks for great investigation! I'm closing this issue

**Comment by tiangolo** [comment]
Thanks for the help here @ycd and @kernc ! 👏  🙇 

Thanks for reporting back and closing the issue @aliev  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
